Deprecate analysis overview list table

merkushin opened the pull request:
### Changes proposed in this Pull Request

* Deprecate `Sensei_Analysis_Overview_List_Table`

<!-- Add the following only if there is any code that is being deprecated. Please list the replacement function or hook that should be called instead, if applicable. Be sure to also add the "Deprecation" label to this PR. -->
### Deprecated Code

* `Sensei_Analysis_Overview_List_Table`

merkushin committed:
Deprecate analysis overview list table

donnapep reviewed (dismissed):
Looks good, although the linter is not happy. 🙂

merkushin committed:
Fix linter issue and remove deprecation call from constructor

Deprecation call in constructor caused failure of tests.

merkushin commented:
@donnapep fixed linter's issue and removed deprecation call from constructor :( as it caused test failures.

donnapep commented:
I think we'll need to put a deprecation notice somewhere; otherwise people won't be aware that we're going to remove it at some point. Perhaps in all the other functions in this class would be good.

merkushin commented:
@donnapep I'll look if I can fix tests without removing deprecation.

merkushin committed:
Deprecate constructor and delete tests for deprecated class

Deprecation in constructor causes an error and a failure of tests that we can't handle gracefully.

merkushin commented:
@donnapep PR updated. Tests for the deprecated class removed.

donnapep reviewed (approved):
🎉
